src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -49,12 +49,9 @@ public sealed partial class TokenCache : ITokenCacheInternal
             }
 
             var tenantId = TokenResponseHelper.GetTenantId(idToken, requestParams);
-
-            bool isAdfsAuthority = requestParams.AuthorityInfo.AuthorityType == AuthorityType.Adfs;
-            bool isAadAuthority = requestParams.AuthorityInfo.AuthorityType == AuthorityType.Aad;
-            string preferredUsername = TokenResponseHelper.GetPreferredUsernameFromIdToken(isAdfsAuthority, idToken);
-            string username = isAdfsAuthority ? idToken?.Upn : preferredUsername;
```

Comment:
I think there's a behavior change here, don't know if important, though. With old code, `username` can be null, if authority is ADFS, and Idtoken or IdToken.Upn is null. `username` is later added to the Account object. With new code, `username` will never be null, (it'll at least be `NullPreferredUsernameDisplayLabel`)

Reply:
We should be consistent in handling Username. In hindsight `NullPreferredUsernameDisplayLabel` is not a great solution, but at least it's a consistent one :)